src/monmove.cpp

```diff
@@ -1474,12 +1474,23 @@ bool monster::move_to( const tripoint &p, bool force, const float stagger_adjust
     }
 
     if( has_flag( MF_DRIPS_NAPALM ) ) {
-        if( one_in( 10 ) ) {
-            g->m.add_item_or_charges( pos(), item( "napalm" ) );
+        if( true ) { // one_in( 10 ) // removed randomness to test
+            // grab the pressurized tank from the monster
+            // find out how much is in it,
+            int store = ammo["pressurized_tank"];
+            // if it has more napalm, drop some and reduce ammo in tank
+            if( store > 0 ) {
+                g->m.add_item_or_charges( pos(), item( "napalm" ) ); // TODO how to set amount leaked
```

Comment:
Use `map::spawn_item` - it has an explicit `charges` parameter so you can set the amount of spawned napalm. Or create the `item` instance with an explicit charges amount (see constructor of `item` - the `qty` parameter sets its charges).

Reply:
Hey. when spawning with that method, it drops jerry cans instead of the liquid

Reply:
Hey. when spawning with that method, it drops jerry cans instead of the liquid. I got it working with the item() by just adding a number.